### PR TITLE
Several NPC-related matching functions. Modifed Npc struct with new float value.

### PR DESCRIPTION
**Pull request by JoshW-7**

### CODE REVIEW

File: src/code_f2470_len_27f0.c [hunk 1]
```diff
@@ -100,9 +100,45 @@ ApiStatus SetNpcJumpscale(ScriptInstance* script, s32 isInitialCall) {
...
+
+    if (npcPtr != NULL) {
+        set_variable(script, outVar, npcPtr->currentAnim);
+        return ApiStatus_DONE2; // Doesn't match if omitted
+    }
+    return ApiStatus_DONE2;
+}
+
+// TODO: Figure out meaning of float value in Npc struct that this function is setting
+ApiStatus func_802CE1C0(ScriptInstance* script, s32 isInitialCall) {
```
Comment (JoshW-7) [inline src/code_f2470_len_27f0.c:130]: Note: func_802CE1C0 was in GetNpcAnimation.s, which is why there was not a file removed for it.

File: include/common_structs.h [hunk 2]
```diff
@@ -61,7 +61,9 @@ typedef struct Npc {
     /* 0x020 */ struct NpcBlurData* blurData; /* related to movement somehow... */
     /* 0x024 */ char unk_24[4];
     /* 0x028 */ u32 currentAnim;
-    /* 0x02C */ char unk_2C[12];
+    /* 0x02C */ char unk_2C[4];
+    /* 0x030 */ f32 unk_30;
+    /* 0x034 */ char unk_34[4];
     /* 0x038 */ struct Vec3f pos;
```
Comment (JoshW-7) [inline include/common_structs.h:67]: I have a hunch unk_2C and unk_34 are also floats, but I don't know for sure, so they're still char arrays for now.

File: src/code_f2470_len_27f0.c (hunk 1 above)
Comment (nanaian) [reply src/code_f2470_len_27f0.c:130]: This function is only ever used twice ingame - and it's in the Lava Piranha fight on the Kolerado NPC:
image
I imagine it has something to do with the NPC's scale factor...

Comment (JoshW-7) [reply src/code_f2470_len_27f0.c:130]: [Link to part of the game where this occurs](https://youtu.be/pdrhCW7qAZc?t=42505)

Comment (JoshW-7) [reply src/code_f2470_len_27f0.c]: Added commits (below) to change the unknown float value to animationSpeed

File: src/code_f2470_len_27f0.c [hunk 3]
```diff
@@ -36,31 +36,109 @@ INCLUDE_API_ASM(code_f2470_len_27f0, SetNpcPos);
...
+    Bytecode* ptrReadPos = script->ptrReadPos;
     NpcId npcID = get_variable(script, *ptrReadPos++);
     f32 speed = get_float_variable(script, *ptrReadPos);
     Npc* npcPtr = resolve_npc(script, npcID);
 
-    npcPtr->moveSpeed = speed;
     if(npcPtr != NULL) {
-        return ApiStatus_DONE2;
+        npcPtr->moveSpeed = speed;
+        return ApiStatus_DONE2; // Doesn't match if omitted
```
[the remark this answers is not in the document]
Comment (JoshW-7) [reply src/code_f2470_len_27f0.c:85]: The comment is mostly to make it clear for anybody reviewing or modifying later that the function cannot be optimized in that way. I'll remove them though.

**Pull request merged**